In [1]:
import math

Null Hypothesis = Assumes that there is no effect or difference, That the mean systolic blood pressure of the sample is equal to that of the general population

Alternative Hypotheses = That the mean systolic blood pressure of the sample is not equal to that of the general population and not by chance. 

In [3]:
sample_mean = 130.1
sample_std = 21.21
pop_mean = 120
n = 100

statistic = (sample_mean - pop_mean)/(sample_std/math.sqrt(n))
statistic

4.761904761904759

- With a z_score of 4.76 we can easily say that we can reject our null hypothesis. The result suggests that the sample mean is significantly different from the population mean and is unlikly to be due to random chance

# T-test and P-value


H0 = That the new machine is the same speed as the old machine.
H1 = that the new machine is faster than the old machine

from scipy.stats import ttest_ind, norm
import numpy as np
import pandas as pd


In [25]:

import chardet


with open('files_for_lab/machine.txt', 'rb') as file:
    result = chardet.detect(file.read())

df = pd.read_csv('files_for_lab/machine.txt', encoding=result['encoding'], delimiter='\t')

df

New machine      Old machine
0         42.1             42.7
1         41.0             43.6
2         41.3             43.8
3         41.8             43.3
4         42.4             42.5
5         42.8             43.5
6         43.2             43.1
7         42.3             41.7
8         41.8             44.0
9         42.7             44.1

In [36]:
new_machine_mean = np.mean(df['New machine'])
old_machine_mean = np.mean(df['    Old machine'])

print(f'The mean speed of the new machine is {new_machine_mean} compared to the old machine of {old_machine_mean}')

The mean speed of the new machine is 42.14 compared to the old machine of 43.230000000000004


In [37]:
ttest = ttest_ind(df['New machine'], df['    Old machine'])
ttest

Ttest_indResult(statistic=-3.3972307061176026, pvalue=0.0032111425007745158)

#### With a pvalue of 0.0032111425007745158 we can safely say we can rejct the null hypothesis.

## ANOVA

- H0 = The etching process stays the same regardless of changing the power.
- H1 = The etching process improves by changing the power.
- Significance level is 0.05
- Degrees of freedom is equal to 2. the three power levels - 1.
- Error degress of freedom is equal to 12. the number of observations - the number of groups.

In [40]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [43]:
ANOVA_df = pd.read_excel('files_for_lab/anova_lab_data.xlsx')
ANOVA_df

Power   Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [51]:
ANOVA_df.rename(columns={'Etching Rate': 'Etching_Rate', 'Power ' : 'Power' }, inplace=True)

In [52]:
ANOVA_df.groupby('Power').agg(np.mean)

Etching_Rate
Power              
160 W         5.792
180 W         6.238
200 W         8.318

In [53]:
model = ols('Etching_Rate ~ Power', data = ANOVA_df).fit()
sm.stats.anova_lm(model)

df     sum_sq   mean_sq          F    PR(>F)
Power      2.0  18.176653  9.088327  36.878955  0.000008
Residual  12.0   2.957240  0.246437        NaN       NaN

In [55]:
ANOVA_df.pivot(columns = 'Power').describe()

Etching_Rate                    
Power        160 W     180 W     200 W
count     5.000000  5.000000  5.000000
mean      5.792000  6.238000  8.318000
std       0.319875  0.434304  0.669604
min       5.430000  5.660000  7.550000
25%       5.590000  5.980000  7.900000
50%       5.710000  6.240000  8.150000
75%       6.010000  6.600000  8.790000
max       6.220000  6.710000  9.200000

In [59]:
power_a = ANOVA_df[ANOVA_df['Power'] == '200 W']['Etching_Rate']

for power in ANOVA_df['Power'].unique():
    power_b = ANOVA_df[ANOVA_df['Power'] == power]['Etching_Rate']
    print(power, ttest_ind(power_a, power_b))

160 W Ttest_indResult(statistic=7.611403634613074, pvalue=6.237977344615716e-05)
180 W Ttest_indResult(statistic=5.827496614588661, pvalue=0.0003926796476049085)
200 W Ttest_indResult(statistic=0.0, pvalue=1.0)


#### The data strongly suggests that 200 W provides the strongest increase in etching with a mean speed of 8.318 and the anova test for 180 providing a pvalue of less then 0.05 we can easily say 200 W provides the best results and each power level provides different results.